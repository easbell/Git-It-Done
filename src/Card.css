.flashcard {
  background-color: #4F7C8F;
  border-radius: 30px;
  box-shadow: 4px 4px 7px #325561;
  width: 60%;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;
  height: 20rem;
  margin: 0 auto;
  color: #fff;
}