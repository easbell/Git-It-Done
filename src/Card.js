import React, { Component } from 'react';
import Button from './Button.js';
import './Card.css';

export class Card extends Component {
  constructor(props) {
    super(props);

    this.state = {
      cardIndex: 0
    }
  }
  
  correctAnswer = (answer) => {
    const { cardIndex } = this.state
    const { questions, displayMessage } = this.props
    this.setState({ cardIndex: this.state.cardIndex + 1 })
    answer === questions[cardIndex].correctAnswer
      ? displayMessage(true)
      : this.answeredWrong();
  }

  answeredWrong = () => {
    const { cardIndex } = this.state
    const { displayMessage, guessWrong } = this.props
    displayMessage(false)
    guessWrong(this.props.questions[cardIndex])
  }
    
  render() {
    const { cardIndex } = this.state
    return (
    <div className="flashcard">
      <h3>{this.props.questions[cardIndex].prompt}</h3>
      <div>
        <Button 
          answer={this.props.questions[cardIndex].possibleChoices[0]}
          correctAnswer={this.correctAnswer}
        />
        <Button 
          answer={this.props.questions[cardIndex].possibleChoices[1]}
          correctAnswer={this.correctAnswer}
        />
        <Button 
          answer={this.props.questions[cardIndex].possibleChoices[2]}
          correctAnswer={this.correctAnswer}
        />
      </div>
    </div>
    )
  }
}

export default Card
