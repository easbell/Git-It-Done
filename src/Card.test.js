import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import { shallow } from 'enzyme';

const displayMessageMock = jest.fn;
const guessWrongMock = jest.fn;

const mockData = [
  {
    "id": 1,
    "prompt": "Make a directory",
    "possibleChoices": ["cdir", "mkdir", "newdir"],
    "correctAnswer": "mkdir",
    "save": false
  },
  {
    "id": 2,
    "prompt": "Change directory",
    "possibleChoices": ["change", "cngdir", "cd"],
    "correctAnswer": "cd",
    "save": false
  },
  {
    "id": 3,
    "prompt": "Go to previous directory",
    "possibleChoices": ["cd ..", "cd .", "prev .."],
    "correctAnswer": "cd ..",
    "save": false
  }
]


describe('Card', () => {
  const wrapper = shallow(
    <Card 
      questions={mockData}
      displayMessage={displayMessageMock}
      guessWrong={guessWrongMock}
    />
  );

  it('should match snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have proper values for default states', () => {
    expect(wrapper.state()).toEqual({cardIndex: 0});
  });

  it.skip('should invoke the correct function depending on if answer is right or wrong', () => {
    wrapper.instance().correctAnswer('mkdir');
  });

})