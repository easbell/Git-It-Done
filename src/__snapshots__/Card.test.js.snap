// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot with all data passed in 1`] = `
<div
  className="flashcard"
>
  <h3>
    Make a directory
  </h3>
  <div
    className="answer-btns"
  >
    <Button
      answer="cdir"
      correctAnswer={[Function]}
    />
    <Button
      answer="mkdir"
      correctAnswer={[Function]}
    />
    <Button
      answer="newdir"
      correctAnswer={[Function]}
    />
  </div>
</div>
`;
